src/app/components/sort-by/sort-by.test.tsx

```diff
@@ -0,0 +1,150 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import { FlightSortByProps, FlightSortBy } from "./sort-by";
+import { SortBy, SortParameters } from "../../models/flight-parameters";
+
+jest.mock('../../../components/ui/button', () => ({
+  Button: ({ children, ...props }: any) => {
+    return <button {...props}>{...children}</button>;
+  }
+}));
+
+enum ChevronTexts {
+  Down = 'ChevronDown',
+  Up = 'ChevronUp'
+}
+
+jest.mock('lucide-react', () => ({
+  ChevronDownIcon: () => <span>{ChevronTexts.Down}</span>,
+  ChevronUpIcon: () => <span>{ChevronTexts.Up}</span>
+}))
+
+describe(FlightSortBy.name, () => {
+  let props: FlightSortByProps;
+
+  beforeEach(() => {
+    props = {
+      setSortByParams: jest.fn()
+    };
+
+    jest.clearAllMocks();
+  });
+
+  describe('Reset', () => {
+    it('should call reset params when clicked', () => {
+      render(<FlightSortBy {...props}></FlightSortBy>);
+
+      const resetElement = screen.getByText(/reset/i);
+      fireEvent.click(resetElement);
+
+      const params: SortParameters = { ascending: true, sortBy: undefined };
+      expect(props.setSortByParams).toHaveBeenCalledWith(params);
+    });
+  });
+
+  it('should render button corresponding to all possible sort options', () => {
+    render(<FlightSortBy {...props}></FlightSortBy>);
+
+    // don't have to use expect for these since getBy throws error unless there is specifically one matching element
+    screen.getByRole('button', { name: /duration/i });
```

Comment:
what does this test?

Reply:
I did not add assertions since getBy throws an error if there is no match found, but I see that is not readable